core/src/main/java/de/danoeh/antennapod/core/util/Converter.java

```diff
@@ -73,6 +73,20 @@ public static String getDurationStringLong(int duration) {
 
     	return String.format(Locale.getDefault(), "%02d:%02d:%02d", h, m, s);
     }
+
+    /** Convert millisecond according to the current playback speed
+     * @param time: time to convert
+     * @param speed: current playback speed
+     * @param timeRespectSpeed: if false no conversion is done, the time is simply returned
+     * @return converted time (can be < 0 if time is < 0)
+     */
+    public static int convertTimeWithSpeed(int time, float speed, boolean timeRespectSpeed) {
```

Comment:
For me, this is a really weird method.
Boolean parameters have always been a code smell to me (if your IDE doesn't show the parameter name, you are basically lost). Does nothing when the third parameter is false and doesn't need the second parameter in these cases...

Maybe the strategy pattern? Initialise the right implementation based on the preference?

Reply:
yeah you're right ...

I tryied something else with a dedicated converter class.
This class is instatiated with the boolean and float. It is possible to call UserPreferences directly (static method) in the class but I don't really like to have class that call other without declaring them as parameter. I can change this though

The converter class need to be reinstatiated everytime the parameter (speed) change.

Let me know if you have something better in mind...